domain/secret/service/service.go

```diff
@@ -721,14 +768,10 @@ func (s *SecretService) getAppOwnedOrUnitOwnedSecretMetadata(ctx context.Context
 		// Should never happen.
 		return nil, errors.Trace(err)
 	}
-	owners := []CharmSecretOwner{{
-		Kind: ApplicationOwner,
-		ID:   appName,
-	}, {
-		Kind: UnitOwner,
-		ID:   unitName,
-	}}
-	metadata, _, err := s.ListCharmSecrets(ctx, owners...)
+	metadata, err := s.secretState.ListCharmSecrets(ctx,
```

Comment:
This new state method doesn't need to return the entire secret metadata struct. The call site just needs url, label, owner. We can use a new domain struct.